src/API/REST/V3/Routes/Donations/DonationController.php

```diff
@@ -447,11 +465,21 @@ public function delete_items($request): WP_REST_Response
                 ->anonymousMode(new DonationAnonymousMode('include'))
                 ->exports();
 
-            if ($donation->delete()) {
-                $deleted[] = ['id' => $id, 'previous' => $item];
-            } else {
-                $errors[] = ['id' => $id, 'message' => __('Failed to delete donation', 'give')];
-            }
+                if ($force) {
+                    if ($donation->delete()) {
+                        $deleted[] = ['id' => $id, 'previous' => $item];
+                    } else {
+                        $errors[] = ['id' => $id, 'message' => __('Failed to delete donation', 'give')];
+                    }
+                } else {
+                    $trashed = wp_trash_post($donation->id);
```

Comment:
I'm hesitant to use `wp_trash_post` directly here as we want this endpoint to interact with the donation model & abstract awa the data layer as much as possible.  In the future when the storage of a donation changes - this would be something we would have to update.

`wp_trash_post` does a couple things:
1. Changes the status to `trash`
2. Stores the previous status & time in meta `_wp_trash_meta_status`, `_wp_trash_meta_time` (as a way to restore to previous status)

All that is something we can support directly in the donation repository with a new `trash()` method and/or build it into the existing delete method.

At the very least if we simply changed the status of the donation to trash and saved - then that would be better for maintainability here.

Reply:
Thanks for the context! That makes sense in terms of long-term maintainability and keeping the data layer abstracted.

I pushed up a commit here to reflect that approach: [59a1f8b](https://github.com/your-repo/commit/59a1f8b43180d54eb857a31e5bd28aa1f95729e1). Let me know if this looks more in line with what you were thinking!